src/LayoutManager/LayoutStrategies/utils.ts

```diff
@@ -0,0 +1,56 @@
+import { Point } from '../../Point';
+import type { Group } from '../../shapes/Group';
+import type { FabricObject } from '../../shapes/Object/FabricObject';
+import type { TMat2D } from '../../typedefs';
+import { cos } from '../../util/misc/cos';
+import { multiplyTransformMatrices, qrDecompose } from '../../util/misc/matrix';
+import { calcPlaneChangeMatrix } from '../../util/misc/planeChange';
+import { degreesToRadians } from '../../util/misc/radiansDegreesConversion';
+import { sin } from '../../util/misc/sin';
+
+/**
+ * @todo this is very buggy => refactor in redo coords
+ */
+export const getObjectSizeVector = (
+  object: FabricObject,
+  /**
+   * account for selected objects in active selection
+   */
+  planeChange?: TMat2D
+) => {
+  const decomposition = planeChange
+    ? qrDecompose(
+        multiplyTransformMatrices(planeChange, object.calcOwnMatrix())
+      )
+    : undefined;
+  const sizeVector = object
+    ._getTransformedDimensions(decomposition)
```

Comment:
One part of my issue with the calcBounding box starts from here.
_getTransformedDimensions should not have options and was a mistake to put them there.
The options object is an any and in this case we are putting down a full decomposition, including an angle.
In the case this happens the _getTransformedDimensions will 'secretly' push the angle down to recompose a matrix that will factor in both the planeChange angle and the object angle, the bounding box will be rotated.

Later down it seems like  we want to compensate this angle we get from sizeVector but we do that only with object.angle

In the case of a planeChange the box includes both angles
in the case without planeChange the box includes NO angles at all

But if the object has an angle we compensate it anyway.

This seems a bug lurking to me.

Can you try to explain why are we compensating the object angle in general? i imagined that it was because we want the bounding box of the object aligned with the group angle, but those calculations do not seems to reflect that

Reply:
There is a bug in this code and I left a comment for.
_getTransformedDimensions is a bad impk regardless of the options.
It can't move planes once calculated, that is why I had tonadd the options.
I will look into what you mentioned but I want to say clearly that this is not worth it.
It must be rewritten, stroke calc is wrong. That is how I started with redoing coords.
Put aside what you thibk of it, it fixes this very issue

Reply:
I read the code again, what do you mean about the angle?
I am not attempting to fix this also because stroke uniform is broken all across fabric so I have no way to reliably measure objects w/o #8767

Reply:
the angle compensation code is hidden behind 
```
qrDecomposition?.angle ?? object.angle
```

that means if we have a qrDecomposition it triggers if the angle of the qr decomposition is not 0, otherwise it doesn't
if the angle of qrDecomposition is different than 0 then we compensate by object.angle.

But this qrDecomposition angle can be different than 0 for a series of reasons:
- planeChange has an angle and object doesn't
- planeChange and object they both have an angle
- planeChange does not have an angle and the object has

OR

they could be both have an angle that compensate to 0, and the plane change includes the angle from both groups that create the the plane change matrix.

I m saying it seems strange that if any angle is present, the compensation has to happen with the object angle.
I wish there were a set of use cases to try

Reply:
> There is a bug in this code and I left a comment for.

Do you remember where this comment his? or what the bug is?

Reply:
i split the code in this example for better discussion

```ts
const getObjectSizeVector = (object: FabricObject) => {
  const sizeVector = object._getTransformedDimensions().scalarDivide(2);
  if (object.angle) {
    return compensateAngle(object.angle, sizeVector);
  }
  return sizeVector;
};
```

```ts
const getObjectSizeVectorWithPlaneChange = (
  object: FabricObject,
  /**
   * account for selected objects in active selection
   */
  planeChange: TMat2D
) => {
  const decomposition = qrDecompose(
    multiplyTransformMatrices(planeChange, object.calcOwnMatrix())
  );
  const sizeVector = object
    ._getTransformedDimensions(decomposition)
    .scalarDivide(2);
  if (decomposition.angle !== 0) {
    return compensateAngle(object.angle, sizeVector);
  }
  return sizeVector;
};
```

this is how i read the 2 different codepath.
The part in which we do 
```
  if (decomposition.angle !== 0) {
    return compensateAngle(object.angle, sizeVector);
  }
```
is unclear to me.

The first question i made and of which i m not sure yet, is if this code is supposed to counter rotate the size vector to repsect the current group angle

Reply:
In the case with no plane change, you get sizeVector that is not influenced by object angle, and then you eventually apply/remove object angle from it.

in the case with the plane change, you get size vector that is influenced by all the angles, group, target group and object itself and then only if the resulting angle from all the angles is different from 0, you apply /remove object angle from it.

In case 2 the size vector is already influenced by object angle in some way

Reply:
<img width="772" alt="image" src="https://github.com/fabricjs/fabric.js/assets/1194048/ec2a1c4f-481a-4a2e-9c9a-815f4c72894b">

There is something that i don't get on how this calculation wants to work. It tries to reduce to 2 points something that before we were doing with 4 points.
I tried to add some debug code to rendering in order to visualize what happens and i see that while the result is correct, the logic behind it is unclear.

Reply:
Always we have the issue of strokeUniform I didn't address here because I don't think it is stable enough in the rest of the code

Reply:
yeah forget strokeUniform.
_getTransformedDimensions does account for object angle if used improperly.
What i am pointing out is that in one case it is accounting for angle, and for another it isn't.

When you do `getTransformedDimensions(decomposition)` is accounting for the decomposition angle, object and plane change matrix. while when you do `getTransformedDimensions()` it is not.

The issue lies in the destructuring of the options here:

```
  _getTransformedDimensions(options: any = {}): Point {
    const dimOptions = {
      scaleX: this.scaleX,
      scaleY: this.scaleY,
      skewX: this.skewX,
      skewY: this.skewY,
      width: this.width,
      height: this.height,
      strokeWidth: this.strokeWidth,
      ...options,
    };
```
the angle from options is being put back in dimOptions, that then is used to transform the points.

And i wouldn't even be worried about that, the fact is that we are doing that only in one case of 3.

This is complicated and i wanted to talk about it. i got a grasp of this PR since  i m looking at it since long, and there are things i want to talk about

Reply:
But yes in general i really prefer sticking to the 4 point system to create the bounding box rather than those 2 points. The main issue is that i can't relate the code to anything i know about algebra/geometry and so i tend to prefer something that i think is more common knowledge

Reply:
> the angle from options is being put back in dimOptions, that then is used to transform the points.
And i wouldn't even be worried about that, the fact is that we are doing that only in one case of 3.

I don't get this. Let's talk.

Reply:
`dimOptions` does not do anything with the angle

Reply:
ok ok is true, dimOptions takes the angle but then it creates a matrix that doesn't take in account of the angle. i was sure it did.

The picture is explanative and needs to go in the docs somehow. i ll commit it to this PR. What did you use to draw it?

compensate angle is for sure a wrong name, i knew when it was all addition that it wasn't a rotation, but here and in the original PR there wasn't an explanation for that logic, and i couldn't make one up.

I still think the 4 points are easier to manage, but i still do not understand, what is the scope of this calculation. It get me the bounding box, and if you don't say anything else i would argue is the same as getBoundingRect(absolute=true)? So i was asking if this bounding box is actually counter rotated as the group, but i can't see that in the code.

On top of that I still think there is an issue, because if the group is rotated -30 and the object is rotated 30, the decomposition has an angle of 0, but the object is rotated anyway, doesn't it count in that case?

Maybe i have to try how this behave when adding an object to a rotated group and how it gets there.

Reply:
I used to kitchensink - it was too tough
I can commit svg to the readme but then I will make it nicer
I don't mind, but we can expose a planechange matrix with a good name and then it become useful and comprehensive

Reply:
though this is correct something was wrong  in the code, refactored to a simpler solution